### DISCUSSION TITLE
自己用Nginx配置的websocket+TLS总是400

### DISCUSSION THREAD

**Discussion by fakebear** [XTLS/Xray-core#1421] [Q&A]
各位大师，我的nginx已经配好一个https网站，证书均已正常工作，https网页也可以正常访问，所用端口为8080，在这个基础上增加v2ray websocket部分，但总是不成功，从nginx的access log上看直接返回400了，根本没有到v2ray去，所以v2ray log没有任何信息。
我的思路是，在访问预设路径时，nginx直接proxy_pass http://127.0.0.1:49478 （v2ray监听端口），不知道这个思路是不是正确，请教一下各位专家指导一下？谢谢

nginx 8080端口部分的配置（隐藏部分信息）：
```
server {
	listen 8080 ssl;
	listen [::]:8080 ssl;
	
	root /var/www/html;
	index index.html index.htm index.php index.nginx-debian.html;

	server_name xxxxxxxx.xyz;
	
	ssl_certificate /etc/nginx/ssl/fullchain.cer;
	ssl_certificate_key /etc/nginx/ssl/xxxxxxxx.xyz.key;
	
	location / {
		try_files $uri $uri/ =404;
	}
	access_log /var/log/nginx/access.log;
	error_log /var/log/nginx/error.log;
	
	location /happyhour {
		proxy_redirect off;
		proxy_pass http://127.0.0.1:49478/;
		proxy_http_version 1.1;
		proxy_set_header Upgrade $http_upgrade;
		proxy_set_header Connection "upgrade";
		proxy_set_header Host $http_host;
	}

	location ~ \.php$ {
		include snippets/fastcgi-php.conf;
		fastcgi_pass unix:/var/run/php/php7.4-fpm.sock;
	}
}
```

v2ray server端配置
```
{
	"log": {
		"access": "/var/log/xray/access.log",
		"error": "/var/log/xray/error.log",
		"loglevel": "info"
	},
	"inbounds": [
		 {
			"port": 49478,
			"listen": "127.0.0.1",
			"protocol": "vless",
			"settings": {
				"clients": [
					{
						"id": "xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx",
						"level": 0,
						"email": "xxxxx@gmail.com"
					}
				],
				"decryption": "none"
			},
			"streamSettings": {
				"network": "ws",
				"security": "tls",
				"wsSettings": {
					"path": "/happyhour"
				}
			}
		}
	],
	"outbounds": [
		{
			"protocol": "freedom"
		}
	]
}
```

**Comment by cjqhg2012** [comment]
你v2ray配置文件写错了，给你个正确的参考。
{
  "log": {
    "loglevel": "warning",
    "access": "/var/log/v2ray/access.log",
    "error": "/var/log/v2ray/error.log"
   },
  "inbounds": [{
    "port": 10000,
    "protocol": "vmess",
    "settings": {
      "clients": [
        {
          "id": "UUID",
          "level": 1,
          "alterId": 0
        }
      ]
    },
   "streamSettings": {
        "network": "ws",
        "wsSettings": {
    
          "path":"/PATH"(这里和NGINX里面一致）
        }
      },
    "listen": "127.0.0.1"
  }],
  "outbounds": [{
    "protocol": "freedom",
    "settings": {}
  },{
    "protocol": "blackhole",
    "settings": {},
    "tag": "blocked"
  }],
  "routing": {
    "rules": [
      {
        "type": "field",
        "ip": ["geoip:private"],
        "outboundTag": "blocked"
      }
    ]
  }
}

**Reply by fakebear** [reply]
我换了你的这个配置，access log里是**502错误**了
貌似我们的区别就只有下面这些，貌似没有本质区别。我感觉我的配置是nginx给挡住了
![image](https://user-images.githubusercontent.com/723762/206855685-72b1a29a-67fb-45fb-b4e1-159c27114e8b.png)

**Comment by picklefan** [comment]
Nginx proxy_pass 的是http， 所以你的xray配置tls要去掉
```
"security": "tls",
```
改为
```
"security": "none",
```
